modules/http/http-curl-compression.c

```diff
@@ -28,7 +28,63 @@ gint8 CURL_COMPRESSION_DEFAULT = CURL_COMPRESSION_UNCOMPRESSED;
 gchar *CURL_COMPRESSION_LITERAL_ALL = "all";
 gchar *curl_compression_types[] = {"identity", "gzip", "deflate", "brotli", "zstd"};
 
-gboolean http_dd_compress_string(GString *compression_destination, const GString *message, const gint compression)
+gchar *_compression_error_message = "Failed due to %s error. Sending uncompressed data.";
+static inline void _handle_compression_error(GString *compression_dest, gchar *error_description)
+{
+}
+static inline gboolean _raise_compression_status(GString *compression_dest, int algorithm_exit)
 {
   return FALSE;
+}
+
+int _gzip_string(GString *compressed, const GString *message);
+int _deflate_string(GString *compressed, const GString *message);
+int _brotli_string(GString *compressed, const GString *message);
+int _zstd_string(GString *compressed, const GString *message);
+
+gboolean http_dd_compress_string(GString *compression_destination, const GString *message, const gint compression)
+{
+  int err_compr;
```

Comment:
If I understand correctly this will be basically an enum with extra steps. It would be better to introduce the result codes with properly named type. See `LogThreadedResult` in `lib/logthrdest/logthrdestdrv.h`.

Reply:
Could you ellaborate, please?

Reply:
You need a multi-value result code.

Your current implementation uses `int` as its type. With this the signatures of the functions do not show the real result type.
A cleaner implementation would define the result type as an enum, which makes the signatures more on point.

There are good examples of the enum style implementation in the code (`LogThreadedResult`, `_LogMessageValueType`). Please consider implementing this like them.

---

Again, sure this works, but we try to conform to cleaner code than old C standards does.

Reply:
You are referring to `_gzip_string()` and `_brotli_string()`, right? In those cases it might make sense to return with an enum value, which would enable changing from macros to an enum (`_COMPRESSION_...`).
In the case of `http_dd_compress_string` I wanted to hide the true error case, as that is handled internally, adhering to SoC.

Reply:
Done

Reply:
Please use the enum type on the user side, too:
```
-  gint8 message_compression;
+  CurlCompressionType message_compression;
```

For this, you need to typedef it:
```
-enum CurlCompressionTypes
+typedef enum CurlCompressionType_
 {
   CURL_COMPRESSION_UNCOMPRESSED = 0,
   CURL_COMPRESSION_GZIP = 1,
   CURL_COMPRESSION_DEFLATE = 2
-};
+} CurlCompressionType;
```